src/components/ArrayInput/index.js

```diff
@@ -0,0 +1,501 @@
+import utils from '../../helpers/utils';
+import './style.scss';
+
+import Element from '../Element';
+import Container from '../Container';
+import Panel from '../Panel';
+import NumericInput from '../NumericInput';
+import Button from '../Button';
+
+const CLASS_ARRAY_INPUT = 'pcui-array-input';
+const CLASS_ARRAY_EMPTY = 'pcui-array-empty';
+const CLASS_ARRAY_SIZE = CLASS_ARRAY_INPUT + '-size';
+const CLASS_ARRAY_CONTAINER = CLASS_ARRAY_INPUT + '-items';
+const CLASS_ARRAY_ELEMENT = CLASS_ARRAY_INPUT + '-item';
+const CLASS_ARRAY_DELETE = CLASS_ARRAY_ELEMENT + '-delete';
+
+var DEFAULTS = {
+    boolean: false,
+    number: 0,
+    string: '',
+    vec2: [0, 0],
+    vec3: [0, 0, 0],
+    vec4: [0, 0, 0, 0]
+};
+
+/**
+ * @event
+ * @name ArrayInput#linkElement
+ * @property {Element} element The array element
+ * @property {number} index The index of the array element
+ * @property {string} path The path linked
+ * @description Fired when an array element is linked to observers
+ */
+
+/**
+ * @event
+ * @name ArrayInput#unlinkElement
+ * @property {Element} element The array element
+ * @property {number} index The index of the array element
+ * @description Fired when an array element is unlinked from observers
+ */
+
+/**
+ * @name ArrayInput
+ * @classdesc Element that allows editing an array of values.
+ * @property {boolean} renderChanges If true the input will flash when changed.
+ * @property {boolean} fixedSize If true then editing the number of elements that the array has will not be allowed.
+ * @property {object} elementArgs Arguments for each array Element.
+ * @property {object} inputClass The class that should be used as the input for each array element. Can be one of: TextInput, BooleanInput, AssetInput, VectorInput
+ * @augments Element
```

Comment:
We should also add the following:
`@mixes IBindable`
`@mixes IFocusable`

Reply:
Cheers, i'll add these in 👍